zookeeper/core/component.py

```diff
@@ -304,6 +302,52 @@ def wrapped_fn(instance) -> List[str]:
     component_cls.__dir__ = wrapped_fn
 
 
+def _wrap_configure(component_cls: Type) -> None:
+    if hasattr(component_cls, "__configure__"):
+        if not callable(component_cls.__configure__):
+            raise TypeError(
+                "The `__configure__` attribute of a @component class must be a "
+                "method."
+            )
+        call_args = inspect.signature(component_cls.__configure__).parameters
+        configure_args = inspect.signature(configure).parameters
+
+        for arg_name, arg_param in call_args.items():
+            if (
+                arg_param.kind in (arg_param.VAR_POSITIONAL, arg_param.VAR_KEYWORD)
+                or arg_name in configure_args
+                or arg_name == "self"
+            ):
+                continue
+            raise TypeError(
+                "The `__configure__` method of a @component class must match the "
+                f"arguments of `configure()`, but `{component_cls.__name__}.__configure__`"
+                f" accepts arguments {tuple(name for name in call_args)}. Valid "
```

Comment:
This should work as well
```suggestion
                f" accepts arguments {tuple(call_args)}. Valid "
```

Reply:
It doesn't, because `call_args` is a dictionary (or some related type). But I guess we can just do `tuple(call_args.keys())` or something, let me try

Reply:
This should also work for dictionaries:
```python
assert tuple({"a": 1, "b": 2, "c": 3}) == tuple({"a": 1, "b": 2, "c": 3}.keys()) == ("a", "b", "c")
```

Reply:
Oh damn :exploding_head: